core/src/main/java/org/apache/accumulo/core/clientImpl/ClientTabletCache.java

```diff
@@ -328,6 +338,10 @@ public Optional<String> getTserverLocation() {
     public Optional<String> getTserverSession() {
       return Optional.ofNullable(tserverSession);
     }
+
+    public TabletHostingGoal getGoal() {
```

Comment:
I think this goal in the cache could be really useful in the future, but its a bit different than the location so tried to capture that in the javadoc.

```suggestion
    /**
      * This information in the cache may have a higher chance of being stale than other entries in the cache.
      * The location in the cache is cleared when something tries to use the location and its wrong.  There is 
      * currently no feedback loop like that for the cached goal.  If this information is needed to be fresh may want
      * to consider clearing the cache.
      */
    public TabletHostingGoal getGoal() {
```

Since its a bit different we may want to know its age when using it from the cache.  Thinking we could add an age getter to the cachedtablet class.  Could record nano time when its the object is created and then have something like.

```java
   Duration getAge() {
      return Duration.ofNanos(System.nanoTime() - creationTime):
   }
```